### PR TITLE
Kill processes better, and add convenience flags/traces for debugging

### PR DESCRIPTION
**Pull request by alanefl**
Adds better logic to run and cleanup experiment commands. 

- Adds the `bw48.mahi` full-length file, since we're gonna use that from now on
- Adds "tiny" versions of the `bw48.mahi` and Verizon LTE up and down short that only last 5 seconds for quick testing
- Adds the `--tiny-trace` command-line flag to run with a 5-second trace for quick and clean debug iterations (tiny-trace simply uses the tiny versions of the trace files above)
- Adds patch to run LEDBAT from the pantheon indefinitely (as long as the trace file), instead of only 75 seconds
- Refactors the way in which experiment commands are passed from a `CCProtocol` object to the `experiment.py` script: before it was simply a list of strings.  Now it is a more descriptive Ordered Dictionary so that the `experiment.py` script can enforce certain logic for each type of command.  Currently, all "prep" commands are run in the background, and everything other command is run in sequence.   This is the correct logic for all protocols at hand as of this PR.
- Redirects STDOUT and STDERR of all spawned experiment processes to devnull to avoid polluting the terminal
- Adds `--verbose` flag to the `experiments.py` script.  If given, it will simply print out the experiment commands being run. 
- Updates the `start-tcp.sh` script to run `iperf` indefinitely (as long as the trace file), instead of manually 140 seconds
- Adds results and logs to dedicated per-protocol subdirectories under `logs/` and `results/` for clarity.

I tested that the processes are killed between experiments correctly by running:

```bash
python experiment.py --experiment figure2a --tiny-trace --verbose --schemes abc copa cubic cubiccodel cubicpie bbr vegas ledbat pcc quic sprout verus
```

NOTE: to run the above, you need to run the `setup/setup.sh` script again to install stuff for the other Pantheon experiments. 

NOTE: QUIC and SPROUT still have problems. 

NOTE: we need to incorporate looking at the downlink log for protocols that run the sender first: (QUIC, Verus

NOTE: something I did in this PR fixed an error I was having with LEDBAT (maybe a lingering process error?).  Now results from ledbat look like this 

`  ~~ Results for protocol: ledbat ~~ 
        utilization: 87.25%
        throughput: 5.2
        signal delay: 392.0
        queuing delay: 207.0
        power score: 13.2653061224
`

### CODE REVIEW

File: experiment.py [hunk 1]
```diff
@@ -93,24 +157,27 @@ def run_fig2_exp(schemes, args, run_full):
...
-            for c in cmds:
-                print("$ %s" % ' '.join(c.split(' ')))
-                call(c, shell=True)
+           run_cmds(cmds, args.verbose)
         else:
             print(" Experiment skipped ")
 
         print_fig2_results(protocol)
-        print(" ---- Done ---- \n")
+        time.sleep(2)
```
Comment (sktollman) [inline experiment.py:178]: Why sleep?

File: start-tcp.sh [hunk 2]
```diff
@@ -3,7 +3,4 @@ if [ $1 = "bbr" ] ; then
 	sudo tc qdisc add dev ingress root fq pacing
 	sudo tc qdisc show dev ingress
 fi
-iperf -c $MAHIMAHI_BASE -p 42425 -w 16m -t 140 -Z $1 &
-sleep 140
-killall iperf
-killall iperf
+iperf -c $MAHIMAHI_BASE -p 42425 -w 16m -t 1000 -Z $1
```
Comment (sktollman) [inline start-tcp.sh:6]: Why do we no longer need to run processes in the background?

**Review by sktollman — APPROVED** [review]
Generally looks good, thanks for doing this! Reiterating that squashing commits can make it easier for the reviewer to see what's happening and makes for a cleaner commit history in master :)

File: experiment.py (hunk 1 above)
Comment (alanefl) [reply experiment.py:178]: Safety measure between experiments.  I figure that if some ports need to be cleared after the processes that used them have been killed, etc, giving some small time between experiments may be good practice

File: start-tcp.sh (hunk 2 above)
Comment (alanefl) [reply start-tcp.sh:6]: This command is run inside a mahimahi shell, and we wait for that process to finish.  The shell will exit once the trace has been played once, so this should stay in the foreground

**Pull request merged**